### PR TITLE
iOS Support

### LINKED ISSUE #100: iOS Support
I noticed this issue on [glutin](https://github.com/tomaka/glutin/issues/29), and I think it applies to winit as well. I didn't see it get copied over, however, the build is definitely still broken. I'm currently working on a fork that fixes the build and adds full iOS support.

### PR DESCRIPTION
**Pull request by cjhowe**
WIP

- [x] Fix iOS build
- [x] Basic support
- [ ] Events
- [ ] Implement all platform::Window methods

(copied from tomaka/glutin#29)
Fixes #100

### CODE REVIEW

File: src/platform/ios/mod.rs [hunk 1]
```diff
@@ -1,54 +1,495 @@
...
+//!
+//! fn start_inner() {
+//!    ...
+//! }
+//!
+//! ```
+//!
+//! Compile project and then drag resulting .a into Xcode project. Add glutin.h to xcode.
+//!
+//! ```c
```
Comment (tomaka) [inline src/platform/ios/mod.rs]: I don't think 'c' works. Should be 'ignore'.

File: src/platform/ios/mod.rs [hunk 2]
```diff
@@ -1,54 +1,495 @@
...
+//!  - applicationWillEnterForeground is Suspended(false)
+//!  - applicationWillTerminate is Closed
+//!
+//! Keep in mind that after Closed event is received every attempt to draw with
+//! opengl will result in segfault.
+//!
+//! Also note that app will not receive Closed event if suspended, it will be SIGKILL'ed
+
 #![cfg(target_os = "ios")]
+#![deny(warnings)]
```
Comment (tomaka) [inline src/platform/ios/mod.rs]: Please don't use `deny(warnings)`. The Rust core team tends to not care about adding warnings whenever they want, which will in turn break the build.

**Comment by cjhowe** [comment]
I was able to compile and run a small example on iOS. It seems like most of what's there is working, but a lot of functionality is still missing. I'll start working on that this weekend.

**Comment by cjhowe** [comment]
I tried this Travis config with `rustup.rs` before I figured out how to install a new target with `rustup.sh`. Using `rustup.sh` for the Travis config is faster, since it will reuse the installation that Travis makes. For documentation, I wanted to leave the configuration using `rustup.rs` here. I'm going to remove those commits.

```
language: rust

rust:
  - nightly
  - stable

cache: cargo

addons:
  apt:
    packages:
    - libxxf86vm-dev

install:
  - curl https://sh.rustup.rs -sSf | sh -s -- -y
  - export PATH=~/.cargo/bin/:$PATH
  - rustup default $TRAVIS_RUST_VERSION
  - if [ $TRAVIS_OS_NAME = osx ]; then rustup target add x86_64-apple-ios; fi

script:
  - cargo build --verbose
  - if [ $TRAVIS_OS_NAME = osx ]; then cargo build --target x86_64-apple-ios --verbose; fi
  - cargo test --verbose

os:
  - linux
  - osx

after_success:
  - |
      [ $TRAVIS_BRANCH = master ] &&
      [ $TRAVIS_PULL_REQUEST = false ] &&
      cargo publish --token ${CRATESIO_TOKEN}
```

**Comment by cjhowe** [comment]
Yeah, it doesn't look like touch input ever worked. It also looks like some of the FFI code is replicating work from [this crate](https://github.com/SSheldon/rust-objc-foundation). I may make some contributions to that library if necessary before finishing this pull request. I should have a lot more time to work on this starting now, so expect to see progress in the near future.

**Comment by cjhowe** [comment]
I have stopped working on this. I'm not sure if I'll be able to come back to it, but maybe someone can continue the work I started here. It turns out my original project idea that relied on this isn't feasible. Most of the work so far has been to clean up the old code and get it to compile and work on Travis, so we could merge it as-is to let someone else build off this later.

**Comment by tomaka** [comment]
Well, let's merge this then. Thanks for the work!

**Pull request merged**